src/singletons.js

```diff
@@ -24,7 +29,19 @@ async function ERC1820Registry (funder) {
 
   await web3.eth.sendSignedTransaction(ERC1820_REGISTRY_DEPLOY_TX);
 
-  return registry.clone();
+  return getDeployedERC1820Registry();
+}
+
+function getDeployedERC1820Registry () {
+  const environment = require('./configure-environment').getEnviroment();
+
+  if (environment === 'truffle') {
+    return truffleERC1820Registry.at(ERC1820_REGISTRY_ADDRESS);
```

Comment:
I'm pretty sure this `.at` method returned a `Promise` at some point. Is that not so anymore?

Reply:
It does return a promise because it checks if the bytecode matches, but the outer function is already `async`, and this promise is the final return value.

Do you think `return await` is more readable? There is an [eslint rule that forbids this](https://eslint.org/docs/rules/no-return-await), which is why I assume the plain `return` is the preferred style.

Reply:
Oh. I guess plain `return` is better. But note that the containing function is _not_ async, only the one above it. If we're keeping this please make `getDeployedERC1820Registry` async.

However... I think we can avoid the bytecode check by using `new TruffleERC1820Registry(address)` instead of `.at`. (Please capitalize the first character if implementing this change.)

Reply:
The address can be specified when creating the abstraction, but then you need to use `.deployed()`, which also returns a promise.

Making `getDeployedERC1820Registry` `async` makes sense, to prevent errors in the future.